packages/backend/src/app.ts

```diff
@@ -477,19 +482,42 @@ export class App {
     this.openPaymentsServer = koa.listen(port)
   }
 
+  public async startAutoPeeringServer(port: number | string): Promise<void> {
+    const koa = await this.createKoaServer()
+
+    const autoPeeringRoutes = await this.container.use('autoPeeringRoutes')
+    const router = new Router<DefaultState, AppContext>()
+
+    router.use(bodyParser())
+
+    router.get('/', autoPeeringRoutes.get)
+
+    koa.use(router.routes())
+
+    this.autoPeeringServer = koa.listen(port)
+  }
+
   public async shutdown(): Promise<void> {
     return new Promise((resolve): void => {
```

Comment:
I'm a little bit confused by this. Why is the promise getting resolved multiple times instead of just once? If there is an OP server, wouldn't the promise be resolved in that specific `if` block, making any further resolves irrelevant?

Reply:
I've been trying to update this code, but it's behaving weirdly when I update it to be "correct". I wonder why it was built this way originally, tbh.

I'm going to make a separate PR that hopefully fixes this and makes it more clear